File: samples/server/petstore/cpp-restbed/test/java_client_tests/java_client/src/test/java/test_default_unimplemented_server_stubs/StoreApiTest.java
package test_default_unimplemented_server_stubs;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.openapi.example.api.StoreApi;
import org.openapi.example.model.Order;
import org.openapi.example.invoker.ApiClient;
import org.springframework.web.reactive.function.client.WebClientResponseException;

import static helper.ApiClientFactories.setUpStoreApi;
import static org.junit.jupiter.api.Assertions.*;
import static helper.TestingHelper.approveException;

class StoreApiTest {

    StoreApi apiInstance;

    @BeforeEach
    void setUp() {
        apiInstance = setUpStoreApi(1234);
    }

    @Test
    void deleteOrder() {
        WebClientResponseException exception = assertThrows(
                WebClientResponseException.class, () -> {
                    apiInstance.deleteOrder("myOrderId").block();
                });

        approveException(exception);
    }

    @Test
    void getInventory() {
        WebClientResponseException exception = assertThrows(
                WebClientResponseException.class, () -> {
                    apiInstance.getInventory().block();
                });

        approveException(exception);
    }

    @Test
    void getOrderById() {
        WebClientResponseException exception = assertThrows(
                WebClientResponseException.class, () -> {
                    apiInstance.getOrderById(1L).block();
                });

        approveException(exception);
    }

    @Test
    void placeOrder() {
        Order order = new Order();
        WebClientResponseException exception = assertThrows(
                WebClientResponseException.class, () -> {
                    apiInstance.placeOrder(order).block();
                });

        approveException(exception);
    }
}

File: samples/server/petstore/kotlin-spring-default/src/test/kotlin/org/openapitools/api/StoreApiTest.kt
package org.openapitools.api

import org.openapitools.model.Order
import org.junit.jupiter.api.Test
import org.springframework.http.ResponseEntity

class StoreApiTest {

    private val api: StoreApiController = StoreApiController()

    /**
     * To test StoreApiController.deleteOrder
     *
     * @throws ApiException
     *          if the Api call fails
     */
    @Test
    fun deleteOrderTest() {
        val orderId: kotlin.String = TODO()
        val response: ResponseEntity<Unit> = api.deleteOrder(orderId)

        // TODO: test validations
    }

    /**
     * To test StoreApiController.getInventory
     *
     * @throws ApiException
     *          if the Api call fails
     */
    @Test
    fun getInventoryTest() {
        val response: ResponseEntity<Map<String, kotlin.Int>> = api.getInventory()

        // TODO: test validations
    }

    /**
     * To test StoreApiController.getOrderById
     *
     * @throws ApiException
     *          if the Api call fails
     */
    @Test
    fun getOrderByIdTest() {
        val orderId: kotlin.Long = TODO()
        val response: ResponseEntity<Order> = api.getOrderById(orderId)

        // TODO: test validations
    }

    /**
     * To test StoreApiController.placeOrder
     *
     * @throws ApiException
     *          if the Api call fails
     */
    @Test
    fun placeOrderTest() {
        val order: Order = TODO()
        val response: ResponseEntity<Order> = api.placeOrder(order)

        // TODO: test validations
    }
}
